frontend/app/src/components/StatusWidget/StatusWidget.tsx

```diff
@@ -136,237 +106,244 @@ const RUNNING_MAN_DISPLAY_DELAY_TIME_MS = 500
  * Displays various script- and connection-related info: our WebSocket
  * connection status, the run-state of our script, and other transient events.
  */
-class StatusWidget extends PureComponent<StatusWidgetProps, State> {
-  /** onSessionEvent signal connection */
-  private sessionEventConn?: SignalConnection
-
-  private curView?: ReactNode
-
-  private readonly minimizePromptTimer = new Timer()
+const StatusWidget: React.FC<StatusWidgetProps> = ({
+  connectionState,
+  sessionEventDispatcher,
+  scriptRunState,
+  rerunScript,
+  stopScript,
+  allowRunOnSave,
+  //theme,
+}): ReactElement => {
+  const [statusMinimized, setStatusMinimized] = useState(shouldMinimize())
+  const [scriptChangedOnDisk, setScriptChangedOnDisk] = useState(false)
+  const [promptMinimized, setPromptMinimized] = useState(false)
+  const [promptHovered, setPromptHovered] = useState(false)
+  const [showRunningMan, setShowRunningMan] = useState(false)
+  const minimizePromptTimer: React.MutableRefObject<Timer | null> =
+    useRef(null)
+  const showRunningManTimer: React.MutableRefObject<Timer | null> =
+    useRef(null)
+  const sessionEventConn = useRef<SignalConnection>()
+  const theme = useTheme()
+
+  const handleAlwaysRerunClick = useCallback((): void => {
+    if (allowRunOnSave) {
+      rerunScript(true)
+    }
+  }, [allowRunOnSave, rerunScript])
 
-  private readonly showRunningManTimer = new Timer()
+  const handleKeyDown = useCallback(
+    (keyName: string): void => {
+      // NOTE: 'r' is handled at the App Level
+      if (keyName === "a") {
+        handleAlwaysRerunClick()
+      }
+    },
+    [handleAlwaysRerunClick]
+  )
 
-  constructor(props: StatusWidgetProps) {
-    super(props)
+  const isConnected = useCallback((): boolean => {
```

Comment:
probably would want to use a `useMemo` here over `useCallback` but honestly for a boolean comparison, the overhead probs isn't worth

Reply:
Updated this to a variable assignment.